util/archive_handler.go

```diff
@@ -24,38 +24,41 @@ type MtaDescriptor struct {
 // GetMtaDescriptorFromArchive retrieves MTA ID from MTA archive
 func GetMtaDescriptorFromArchive(mtaArchiveFilePath string) (MtaDescriptor, error) {
 	var descriptor MtaDescriptor
-	// Open the mta archive
 	mtaArchiveReader, err := zip.OpenReader(mtaArchiveFilePath)
 	if err != nil {
 		return MtaDescriptor{}, err
 	}
 	defer mtaArchiveReader.Close()
 
-	for _, file := range mtaArchiveReader.File {
-		// Check for the mta descriptor
-		if file.Name == defaultDescriptorLocation {
-
-			descriptorBytes, err := readZipFile(file)
-			if err != nil {
-				return MtaDescriptor{}, err
-			}
+	descriptorFile := findMtaDescriptorFile(mtaArchiveReader.File)
+	if descriptorFile == nil {
+		return MtaDescriptor{}, errors.New("Could not get a valid MTA descriptor from archive")
+	}
 
-			// Unmarshal the content of the temporary deployment descriptor into struct
-			err = yaml.Unmarshal(descriptorBytes, &descriptor)
-			if err != nil {
-				return MtaDescriptor{}, err
-			}
+	descriptorBytes, err := readZipFile(descriptorFile)
+	if err != nil {
+		return MtaDescriptor{}, err
+	}
 
-			// Return the MTA deployment descriptor, if it is valid
-			if descriptor.ID != "" {
-				return descriptor, nil
-			}
-		}
+	if err = yaml.Unmarshal(descriptorBytes, &descriptor); err != nil {
+		return MtaDescriptor{}, err
 	}
 
+	if descriptor.ID != "" {
+		return descriptor, nil
+	}
 	return MtaDescriptor{}, errors.New("Could not get a valid MTA descriptor from archive")
 }
 
+func findMtaDescriptorFile(files []*zip.File) *zip.File {
+	for _, file := range files {
+		if file.Name == defaultDescriptorLocation {
```

Comment:
I know its an old code but seems like comparing directory and file.
Maybe rename "defaultDescriptorLocation" to "deploymentDescriptorFullPath" or something similar?